File: server/models/summary_stat.py
from server.server import db

class SummaryStat(db.Model):
    __tablename__ = 'SummaryStat'
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String, nullable=False)
    entries = db.relationship('SummaryStatEntry', lazy=False)


    @property
    def serialize(self):
        return {
            'id': self.id,
            'name': self.name,
            'entries': [entry.serialize for entry in self.entries] if self.entries is not None else {}
        }

File: client/src/app/models/summary-stat.ts
import { autoserialize, autoserializeAs } from 'cerialize';
import { SummaryStatEntry } from './summary-stat-entry';

export class SummaryStat {
    @autoserialize id: number;
    @autoserialize name: string;
    @autoserializeAs(SummaryStatEntry) entries: SummaryStatEntry[];
}